packages/dynamite/dynamite/lib/src/builder/client.dart

```diff
@@ -272,361 +172,348 @@ Class buildClient(
+
+      for (final responseEntry in responses.entries) {
+        final response = responseEntry.key;
+        final statusCodes = responseEntry.value;
+
+        TypeResult? headersType;
+        if (response.headers != null) {
+          final identifier =
+              '${tag != null ? toDartName(tag, uppercaseFirstCharacter: true) : null}${toDartName(operationId, uppercaseFirstCharacter: true)}Headers';
+          headersType = resolveObject(
+            spec,
+            state,
+            identifier,
+            openapi.Schema(
+              (final b) => b
+                ..properties.replace(
+                  response.headers!.map(
+                    (final headerName, final value) => MapEntry(
+                      headerName.toLowerCase(),
+                      value.schema!,
+                    ),
+                  ),
+                ),
+            ),
+            isHeader: true,
+          );
+        }
 
-          final security = operation.security ?? spec.security ?? BuiltList();
-          final securityRequirements = security.where((final requirement) => requirement.isNotEmpty);
-          final isOptionalSecurity = securityRequirements.length != security.length;
-          code.write('    // coverage:ignore-start\n');
-          for (final requirement in securityRequirements) {
-            final securityScheme = spec.components!.securitySchemes![requirement.keys.single]!;
-            code.write('''
-if (${isRootClient ? 'this' : '_rootClient'}.authentications.where((final a) => a.type == '${securityScheme.type}' && a.scheme == '${securityScheme.scheme}').isNotEmpty) {
-  _headers.addAll(${isRootClient ? 'this' : '_rootClient'}.authentications.singleWhere((final a) => a.type == '${securityScheme.type}' && a.scheme == '${securityScheme.scheme}').headers);
-}
-''');
-            if (securityRequirements.last != requirement) {
-              code.write('else ');
-            }
-          }
-          if (securityRequirements.isNotEmpty && !isOptionalSecurity) {
-            code.write('''
-else {
-  throw Exception('Missing authentication for ${securityRequirements.map((final r) => r.keys.single).join(' or ')}');
+        final dataType = resolveMimeTypeDecode(
+          response,
+          spec,
+          state,
+          toDartName(
+            '$operationId-response${responses.entries.length > 1 ? '-${responses.entries.toList().indexOf(responseEntry)}' : ''}',
+            uppercaseFirstCharacter: true,
+          ),
+        );
+
+        code.writeln(
+          'final _uri = Uri(path: _path, queryParameters: _queryParameters.isNotEmpty ? _queryParameters : null);',
+        );
+
+        if (responses.values.isNotEmpty) {
+          final cases = statusCodes.map((final statusCode) => 'case $statusCode:').join('\n');
+          code.writeln('''
+bool validateStatus(final int statusCode) {
+  switch (statusCode) {
+    $cases
+      return true;
+    default:
+      return false;
+  }
```

Comment:
In theory switch lookups are in constant time while we'd need to iterate over the list potentially being in `n`.
Not that it matters as we don't expect to have more than a few accepting status codes.

Should I change it?

Reply:
Yeah, I think the benefit of cleaner code is way bigger than the small potential performance improvement

Reply:
what do you think of just inlining the validation function with the doRequest call?
Thinking of:

```dart
      response: _rootClient.doRequest(
        'delete',
        uri,
        headers,
        body,
        validateStatus: (final statusCode) {
          switch (statusCode) {
            case 200:
              return true;
            default:
              return false;
          }
        },
      ),
´´´

Reply:
That is better, but I still like the list more for readability reasons. Could you make a quick benchmark to see how much worse the performance will be?

Reply:
It's not even worth measuring as we are talking about nano seconds at most.

Reply:
So let's just go for the list approach?

Reply:
A `Set` fits better, but yes.

Reply:
Let's reignite this discussion:
How do you plan to support status code ranges like `4XX` or `5XX`.

In this case a switch might be better.

Reply:
Another this is default responses.
This would be another great usecase that will be needed in the petstore example.

I think I'll revert it to the switch if that's fine with you

Reply:
Well yeah, for each of the status codes you need to return the respective serializer.
I think for the current case with just returning true/false a switch is overkill.
But once we support multiple different schemas, a switch is probably the only clean way.
Could you already try to implement it this way?

Reply:
OK I'll open an issue as I want this to be done and for us this is just a cosmetic change (for now).

It's probably more needed for the future converter Idea I mentioned above.
I already made some UML on how I'd like it to be but I don't think I'll be able to include it in this PR.

Reply:
I'll keep the set as I already changed to it yesterday